openedx_learning/apps/authoring/publishing/models/publishable_entity.py

```diff
@@ -265,11 +277,47 @@ class Meta:
         verbose_name_plural = "Publishable Entity Versions"
 
 
+class PublishableEntityVersionDependency(models.Model):
+    """
+    Track the PublishableEntities that a PublishableEntityVersion depends on.
+
+    For example, a partcular version of a Unit (U1.v1) might be defined to have
+    unpinned references to Components C1 and C2. That means that any changes in
+    C1 or C2 will affect U1.v1 via DraftSideEffects and PublishedSideEffects. We
+    say that C1 and C2 are dependencies of U1.v1.
+
+    An important restriction is that a PublishableEntityVersion's list of
+    dependencies are defined when the version is created. It is not modified
+    after that. No matter what happens to C1 or C2 (e.g. edit, deletion,
+    un-deletion, reset-draft-version-to-published), they will always be
+    dependencies of U1.v1.
+
+    If someone removes C2 from U1, then that requires creating a new version of
+    U1 (so U2.v2).
+
+    This restriction is important because our ability to calculate and cache the
+    state of "this version of this publishable entity and all its dependencies
+    (children)" relies on this being true.
+
+    .. no_pii
+    """
+    version = models.ForeignKey(PublishableEntityVersion, on_delete=models.RESTRICT)
+    entity = models.ForeignKey(PublishableEntity, on_delete=models.RESTRICT)
```

Comment:
I think at least one of these field names should change so that it's clear what the direction of the dependency is. One idea:

```suggestion
    version = models.ForeignKey(PublishableEntityVersion, on_delete=models.RESTRICT)
    dependent_entity = models.ForeignKey(PublishableEntity, on_delete=models.RESTRICT)
```

Reply:
Yeah, I like that one. I'll do that.